llvm-external-projects/iree-dialects/test/iree_linalgext/convert_to_loops.mlir

```diff
@@ -505,3 +505,90 @@ func @reverse_dim_0(%arg0: memref<?x?xi32>, %arg1: memref<?x?xi32>) {
 // CHECK:             %[[T2:.+]] = arith.subi %[[T1]], %[[I]] : index
 // CHECK:             %[[V0:.+]] = memref.load %[[IN]][%[[I]], %[[J]]]
 // CHECK:             memref.store %[[V0]], %[[OUT]][%[[T2]], %[[J]]] : memref<?x?xi32>
+
+func @scan_1d_inclusive(%0: memref<128xi32>, %1: memref<128xi32>) {
+  iree_linalg_ext.scan dimension(0) inclusive(1) ins(%0 : memref<128xi32>) outs(%1 : memref<128xi32>) {
+    ^bb0(%arg0 : i32, %arg1 : i32):
+      %sum = arith.addi %arg0, %arg1 : i32
+      iree_linalg_ext.yield %sum : i32
+  }
+  return
+}
+// CHECK-LABEL: func @scan_1d_inclusive
+// CHECK-SAME:    %[[BUFI:[a-zA-Z0-9]+]]
+// CHECK-SAME:    %[[BUFO:[a-zA-Z0-9]+]]
+// CHECK-DAG:     %[[C128:.+]] = arith.constant 128 : index
+// CHECK-DAG:     %[[C0:.+]] = arith.constant 0 : index
+// CHECK-DAG:     %[[C1:.+]] = arith.constant 1 : index
+// CHECK:         scf.for %[[ARG1:.+]] = %[[C0]] to %[[C128]] step %[[C1]]
+// CHECK:           %[[COND:.+]] = arith.cmpi eq, %[[ARG1]], %[[C0]] : index
+// CHECK:           scf.if %[[COND]] {
```

Comment:
I'd avoid the `if` if possible and just peel the loop. 

So inclusive would be
```
output[0] = input[0]
for i = 1 to ub step 1 {
   output[i] = output[i-1] + input[i]
}
```

Exclusive would be
```
output[0] = 0
for i = 0 to ub step 1 {
  output[i+1] = output[i] + input[i];
}
```

Reply:
Good point. For now lets just stay with the `scf.if` version you have then. This is only required for last step and hopefully the classic loop rotation transformation in LLVM will do the right thing.

Reply:
Sounds good. thanks!